purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt

```diff
@@ -41,52 +43,60 @@ data class PaywallData(
+    val localizedConfiguration: Pair<Locale, LocalizedConfiguration>
+        @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
         get() {
             val preferredLocales = LocaleListCompat.getDefault()
 
             for (i in 0 until preferredLocales.size()) {
                 preferredLocales.get(i)?.let { locale ->
-                    configForLocale(locale)?.let { localizedConfiguration ->
-                        return localizedConfiguration
+                    // This is a hack to make android studio previews work. In previews, the locale language method
+                    // gives a different result than expected (for example, "es-es" instead of "es").
+                    // In order to fix that, we convert the string to a locale and we parse that as a locale so the
+                    // language and country are correctly parsed.
+                    val localeToCheck = Locale.forLanguageTag(locale.toString())
+                    configForLocale(localeToCheck)?.let { localizedConfiguration ->
+                        return (localeToCheck to localizedConfiguration)
                     }
                 }
             }
 
             return fallbackLocalizedConfiguration
         }
 
-    private val fallbackLocalizedConfiguration: LocalizedConfiguration
+    private val fallbackLocalizedConfiguration: Pair<Locale, LocalizedConfiguration>
+        @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
         get() {
-            return localization.values.first()
+            localization.entries.first().let { localization ->
+                return (getLocaleForLabel(localization.key) to localization.value)
+            }
         }
 
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    private fun getLocaleForLabel(label: String): Locale {
+        // Parsing language requires it to be in the form "en-US", even though Locale.toString() returns "en_US"
+        return Locale.forLanguageTag(label.replace("_", "-"))
+    }
+
     /**
      * @note This allows searching by `Locale` with only language code and missing region (like `en`, `es`, etc).
      *
      * @return [LocalizedConfiguration] for the given [Locale], if found.
      */
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
     fun configForLocale(requiredLocale: Locale): LocalizedConfiguration? {
         return localization[requiredLocale.toString()]
             ?: localization.entries.firstOrNull { (localeKey, _) ->
-                @Suppress("UsePropertyAccessSyntax")
-                Locale(localeKey).languageWithFallback == requiredLocale.languageWithFallback
+                val locale = getLocaleForLabel(localeKey)
+                try {
+                    locale.isO3Language == requiredLocale.isO3Language
+                } catch (e: MissingResourceException) {
+                    errorLog("Locale $this can't obtain ISO3 language code ($e). Falling back to language.")
+                    locale.language == requiredLocale.language
+                }
```

Comment:
This is similar to the iOS version now:
```
private extension Locale {

    func sharesLanguageCode(with other: Locale) -> Bool {
        if #available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, *) {
            return self.language.languageCode == other.language.languageCode
        } else {
            return self.languageCode == other.languageCode
        }
    }

}
```

I'd do the same and make it a method on `Locale` though, something like `Locale.sharesLanguageCodeWith(other: Locale)`?

Reply:
Done in https://github.com/RevenueCat/purchases-android/pull/1267/commits/d01bb5546931c21d7bf4a0bd26d48a44270031ee